features/netsocket/nsapi_dns.cpp

```diff
@@ -147,6 +147,35 @@ extern "C" nsapi_error_t nsapi_dns_add_server(nsapi_addr_t addr, const char *int
     return NSAPI_ERROR_OK;
 }
 
+extern "C" nsapi_error_t nsapi_dns_get_server(NetworkStack *stack, int index, nsapi_addr_t *addr, const char *interface_name)
+{
+    SocketAddress dns_addr;
```

Comment:
This function should call the nsapi_dns_get_server_addr() to get the address so that the functionality is the same as when the DNS query is made and code is not duplicated.
On the call, the total_attempts should be set to value MBED_CONF_NSAPI_DNS_TOTAL_ATTEMPTS so that configuration is taken into account. send_success should be zero. 
Also the other DNS API functions are using SocketAddress as the address (not nsapi_addr_t, e.g. add_dns_server()) so the interface should be updated to use SocketAddress.